distributed/worker.py

```diff
@@ -1929,8 +1937,7 @@ def transition_missing_fetch(
         self._missing_dep_flight.discard(ts)
         ts.state = "fetch"
         ts.done = False
-        self.data_needed.push(ts)
-        return {}, []
+        return self._add_to_data_needed(ts)
```

Comment:
Fixes bug where tasks transitioning from missing to fetch would be ignored by select_keys_for_gather